types/rdfjs__dataset/DatasetCore.d.ts

```diff
@@ -1,12 +1,12 @@
 import * as RDF from 'rdf-js';
 
-interface DatasetCore<InQuad extends RDF.BaseQuad = RDF.Quad> extends RDF.DatasetCore<RDF.Quad, InQuad> {
-    match(subject?: RDF.Term | null, predicate?: RDF.Term | null, object?: RDF.Term | null, graph?: RDF.Term | null): DatasetCore<InQuad>;
+interface DatasetCore<Q extends RDF.BaseQuad = RDF.Quad> extends RDF.DatasetCore<Q, Q> {
+    match(subject?: RDF.Term | null, predicate?: RDF.Term | null, object?: RDF.Term | null, graph?: RDF.Term | null): DatasetCore<Q>;
```

Comment:
Why is match getting redefined here - this is the same as the original definition. Can't you just have

`type DatasetCore<Q extends RDF.BaseQuad = RDF.Quad> = RDF.DatasetCore<Q>`

Note the second type parameter is also not needed because of the defaults in the original package

Reply:
In fact if you are going to change the match definition then why not
```ts
    match(subject?: Q['subject'] | null, predicate?: Q['predicate'] | null, object?: Q['object'] | null, graph?: Q['graph'] | null): DatasetCore<Q>;
```

Reply:
See the return type. It returns its own type and not the rdf-js interface. I think that was the reason. You're right to notice that they are the same now that we don't split the `InQuad` and`OutQuad`

Reply:
Gotta stay interface, however, to avoid redefining all class members by virtue of declaration merging.